### PR TITLE
feat(j-s): add server-resolved originalAncestorId to case response

### PR DESCRIPTION
**Pull request by oddsson**
## What
- Adds an `originalAncestorId` field to the case response, resolved server-side in `CaseInterceptor` so every case carries it.
- Resolution depends on case type: indictment cases return the split case id (`splitCaseId ?? id`); request cases walk the `parentCaseId` chain up to the original ancestor.
- Consolidates this into a single reusable `CaseRepositoryService.findOriginalAncestorId`, now used by both the response interceptor and the police delivery path — replacing the duplicate ancestor walk that lived in `InternalCaseService` and the ad-hoc walk in the interceptor.
- The walk uses a lightweight `findParentCaseId` lookup (selects only the `parentCaseId` column) so it stays cheap to run on every case response.
- Simplifies the web `usePoliceDigitalCaseFile` hook to rely on the server-resolved id (`originalAncestorId ?? caseId`), dropping the previous `originalAncestorId ?? splitCase?.id ?? caseId` fallback chain.
- Adds unit tests covering both routes: indictment (split / non-split) and the request-case parent walk (single-level, multi-level, missing-parent edge case).

## Why
The police digital case file lookup needs the original/ancestor case id, since digital files live on the original case. Previously the frontend stitched this together client-side while the backend kept duplicate ancestor-walk logic in two places. Computing it once on the server gives a single source of truth, keeps the two backend consumers in sync, and lets the frontend drop its fallback chain.

## Screenshots / Gifs
Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review
- [x] AI tools were used in preparing this pull request
      Tools used: Claude Code
- [ ] No AI tools were used in preparing this pull request

## Summary by CodeRabbit

* **New Features**
  * Added `originalAncestorId` field to case data, enabling improved case tracking and ancestry resolution through the API and UI.

* **Bug Fixes**
  * Streamlined case identification logic by centralizing ancestor ID resolution, reducing complexity in parent and split case handling.

* **Tests**
  * Added comprehensive test coverage for ancestor ID resolution, including split cases and multi-level case hierarchies.

### CODE REVIEW

**Review by gudjonss — APPROVED** [review]

File: apps/judicial-system/backend/src/app/modules/repository/services/caseRepository.service.ts [hunk 1]
```diff
@@ -189,6 +190,31 @@ export class CaseRepositoryService {
...
+    }
+
+    // request cases can be extended
+    let originalAncestorId = theCase.id
+    let parentCaseId: string | null | undefined = theCase.parentCaseId
+
+    while (parentCaseId) {
+      originalAncestorId = parentCaseId
+      parentCaseId = await this.findParentCaseId(parentCaseId)
+    }
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/backend/src/app/modules/repository/services/caseRepository.service.ts:213]: We'll assume this will not happen.
